File: src/main/java/io/github/lexcao/tdd/java/reservation/service/impl/ReservationServiceImpl.java
package io.github.lexcao.tdd.java.reservation.service.impl;


import io.github.lexcao.tdd.java.reservation.entity.Reservation;
import io.github.lexcao.tdd.java.reservation.repository.ReservationRepository;
import io.github.lexcao.tdd.java.reservation.service.ReservationService;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class ReservationServiceImpl implements ReservationService {

    private final ReservationRepository repository;

    @Override
    public Reservation makeReservation(Reservation reservation) {
        return repository.save(reservation);
    }
}

File: src/main/kotlin/io/github/lexcao/tdd/reservation/service/impl/ReservationServiceImpl.kt
package io.github.lexcao.tdd.reservation.service.impl

import io.github.lexcao.tdd.reservation.causes.ReservationTimeNotAvailable
import io.github.lexcao.tdd.reservation.entity.Reservation
import io.github.lexcao.tdd.reservation.repository.ReservationRepository
import io.github.lexcao.tdd.reservation.service.ReservationService

class ReservationServiceImpl(private val repository: ReservationRepository) : ReservationService {
    override fun makeReservation(reservation: Reservation): Reservation {
        repository.findByTime(reservation.time)?.run {
            throw ReservationTimeNotAvailable
        }

        return repository.save(reservation)
    }
}